h2o-algos/src/main/java/hex/tree/drf/DRFModel.java

```diff
@@ -55,7 +68,91 @@ public DRFParameters() {
     }
     return preds;
   }
+  
+  @Override
+  public Frame scoreContributions(Frame frame, Key<Frame> destination_key) {
```

Comment:
ScoreContributionsTask and this function are very similar to the one in GBM, can we avoid the code duplication?

Reply:
Yes, the main difference is how the contributions are calculated in the `map()` task for `ScoreContributionsTask`.

Reply:
I have moved ScoreContributionsTask into its own file and scoreContributions into SharedTreeModel.